Runtimes/Core/CMakeLists.txt

```diff
@@ -124,6 +124,11 @@ add_compile_definitions(
   $<$<BOOL:${SwiftCore_ENABLE_UNICODE_DATA}>:-DSWIFT_STDLIB_ENABLE_UNICODE_DATA> # Stubs
   $<$<BOOL:${SwiftCore_ENABLE_ENVIRONMENT}>:-DSWIFT_STDLIB_HAS_ENVIRON> # Concurrency, runtime, shims, platform overlay
   $<$<BOOL:${SwiftCore_ENABLE_BACKDEPLOYMENT_SUPPORT}>:-DSWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT> # Concurrency, Compatibility override, magic symbols
+  # Target libraries that include libDemangling must define the name to use for
+  # the inline namespace to distinguish symbols from those built for the
+  # compiler, in order to avoid possible ODR violations if both are statically
+  # linked into the same binary. (see also commit message for 5b1daa9055c99904c84862ecc313641fd9b26e63)
+  $<$<COMPILE_LANGUAGE:C,CXX>:-DSWIFT_INLINE_NAMESPACE=__runtime> # Demangling, runtime, Concurrency, RemoteInspection, SwiftRemoteMirror, swift-reflection-test
```

Comment:
Since this is only needed by things that pull in `libDemangling`, we should make this a `PUBLIC` define coming from the libDemangling library. Then anything that links Demangling will pick up the appropriate setting.

Reply:
Thanks for the guidance -- I admit I blindly matched the existing code, without seeing its underlying purpose and translating it to proper modern CMake patterns.

Reply:
This global stash of definitions should eventually have the definition attached to the appropriate libraries. This was mostly for expedience when deciphering the logic in `add_swift_target_library` was taking too long.